vertica_python/vertica/messages/backend_messages/notice_response.py

```diff
@@ -63,22 +63,26 @@ class NoticeResponse(BackendMessage):
     def __init__(self, data):
         BackendMessage.__init__(self)
         self.values = {}
+        self.attributes = {}
```

Comment:
What's the difference between `self.values` and `self.attributes` besides the name of keys? Why do we must add `self.attributes`?

Reply:
`self.values` keys is for humans. It contains spaces and uppercases.
Compare `self.values['Error Code']` and `self.attributes['error_code']`.

Yes, it duplicate `setattr(self, attr, value)` but `self` contains too much garbage to pass it as an Exceptions argument directly.